src/Spago/Build.hs

```diff
@@ -179,42 +179,49 @@ repl cacheFlag newPackages sourcePaths pursArgs depsOnly = do
 -- | Test the project: compile and run "Test.Main"
 --   (or the provided module name) with node
 test :: Spago m => Maybe Purs.ModuleName -> BuildOptions -> [Purs.ExtraArg] -> m ()
-test = runWithNode (Purs.ModuleName "Test.Main") (Just "Tests succeeded.") "Tests failed: "
+test = runBackend Nothing (Purs.ModuleName "Test.Main") (Just "Tests succeeded.") "Tests failed: "
 
 -- | Run the project: compile and run "Main"
 --   (or the provided module name) with node
 run :: Spago m => Maybe Purs.ModuleName -> BuildOptions -> [Purs.ExtraArg] -> m ()
-run = runWithNode (Purs.ModuleName "Main") Nothing "Running failed, exit code: "
+run maybeModuleName buildOpts extraArgs = do
+  Config.Config { alternateBackend } <- Config.ensureConfig
+  runBackend alternateBackend (Purs.ModuleName "Main") Nothing "Running failed, exit code: " maybeModuleName buildOpts extraArgs
 
 -- | Run the project with node: compile and run with the provided ModuleName
 --   (or the default one if that's missing)
-runWithNode
+runBackend
   :: Spago m
-  => Purs.ModuleName
+  => Maybe Text
+  -> Purs.ModuleName
   -> Maybe Text
   -> Text
   -> Maybe Purs.ModuleName
   -> BuildOptions
   -> [Purs.ExtraArg]
   -> m ()
-runWithNode defaultModuleName maybeSuccessMessage failureMessage maybeModuleName buildOpts nodeArgs = do
-  echoDebug "Running NodeJS"
-  outputPath <- getOutputPath buildOpts
-  build buildOpts (Just (nodeAction outputPath))
+runBackend maybeBackend defaultModuleName maybeSuccessMessage failureMessage maybeModuleName buildOpts extraArgs = do
+  echoDebug $ "Running with backend: " <> fromMaybe "nodejs" maybeBackend
+  let postBuild = maybe (nodeAction =<< getOutputPath buildOpts) backendAction maybeBackend
+  build buildOpts (Just postBuild)
   where
     moduleName = fromMaybe defaultModuleName maybeModuleName
-    args = Text.intercalate " " $ map Purs.unExtraArg nodeArgs
-    contents outputPath' =
-         let path = fromMaybe "output" outputPath'
+    nodeArgs = Text.intercalate " " $ map Purs.unExtraArg extraArgs
+    nodeContents = \outputPath'
+      -> let path = fromMaybe "output" outputPath'
          in "#!/usr/bin/env node\n\n" <> "require('../" <> Text.pack path <> "/" <> Purs.unModuleName moduleName <> "').main()"
-    cmd = "node .spago/run.js " <> args
+    nodeCmd = "node .spago/run.js " <> nodeArgs
     nodeAction outputPath' = do
-      echoDebug "Writing .spago/run.js"
-      writeTextFile ".spago/run.js" (contents outputPath')
+      echoDebug $ "Writing .spago/run.js"
+      writeTextFile ".spago/run.js" (nodeContents outputPath')
       chmod executable ".spago/run.js"
-      shell cmd empty >>= \case
-        ExitSuccess   -> maybe (pure ()) echo maybeSuccessMessage
+      shell nodeCmd empty >>= \case
+        ExitSuccess   -> fromMaybe (pure ()) (echo <$> maybeSuccessMessage)
         ExitFailure n -> die $ failureMessage <> repr n
+    backendAction backend = do
+      Turtle.proc backend ["--run"] empty >>= \case
```

Comment:
passing `["--run", moduleName]`  (or similar) as an argument to be awesome

Reply:
I can do that but I find passing "Main" to psgo currently doesn't work. Should we first agree on module names for run and test actions?

Reply:
At the moment `pskt` does not support that either. That addition can probably wait for #434 to be resolved. But it's good to know, that the change would not be too hard :)